csp_billing_adapter_microsoft/plugin.py

```diff
@@ -81,55 +80,76 @@ def get_account_info(config: Config):
     The information contains the metadata for compute and network.
     """
     account_info = _get_metadata()
-    account_info['compute'] = json.loads(account_info.get('compute', '{}'))
-    account_info['network'] = json.loads(account_info.get('network', '{}'))
     account_info['cloud_provider'] = get_csp_name(config)
 
     return account_info
 
 
-def get_api_header():
-    """
-    Get the header to be used in requests
-
-    Prefer IMDS which requires a token.
-    """
-    request = urllib.request.Request(
-        METADATA_TOKEN_URL,
-        headers={'Metadata': 'True'},
-        method='GET'
+def _get_metadata():
+    metadata = {}
+    try:
+        metadata = _get_signature()
+        metadata['compute'] = _get_compute_metadata()
+    except ValueError as error:
+        log.error('Could not load JSON from metadata: {}'.format(error))
+        for key in ['compute', 'signature']:
+            if not metadata.get(key):
+                metadata[key] = {}
+
+    return metadata
+
+
+def _get_compute_metadata():
+    compute = json.loads(
+        _fetch_metadata(
+            COMPUTE_URL.format(latest=api_latest_version)
+        )
     )
 
-    try:
-        token = urllib.request.urlopen(request).read().decode()
-    except urllib.error.URLError as error:
-        log.error(f'Failed to retrieve metadata token: {str(error)}')
-        return {}
+    return {
+        'offer': compute.get('offer'),
+        'location': compute.get('location'),
+        'resource_id': compute.get('resourceId')
+    }
 
-    return token
 
+def _get_signature():
+    api_latest_version = _get_latest_api_version(True)
```

Comment:
Should we also add this to _get_compute_metadata() for consistency?

Reply:
Yes, I noticed it testing yesterday, I added the [version](https://github.com/SUSE-Enceladus/csp-billing-adapter-microsoft/pull/4/files#diff-5c007e587ee2c3d8842e12b046658d2904d4a9ad77f39da931d03a4bb2009061R105) but not declared the variable